src/Storages/ColumnsDescription.cpp

```diff
@@ -308,6 +308,12 @@ void ColumnsDescription::flattenNested()
 {
     for (auto it = columns.begin(); it != columns.end();)
     {
+        if (!isNested(it->type))
```

Comment:
Seems it could break a backup restore if backup was created on an older version without this fix, but restored on a newer version with this fix, because `RestorerFromBackup::checkTable` will fail. (Not sure 100%, but better to check and add a test).

Reply:
Check https://github.com/ClickHouse/ClickHouse/pull/56306/files#diff-4de47ebf06a132879e6b724fa2f2b2e687eb42c544f956613c1961dc7921917cR10.
but modify settings a bit. In this test I backup a table with `flatten_nested=0` and restore with `flatten_nested=1`, but instead create backup with `flatten_nested=1` as well. It seems it can fail.